x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/formula/validation.ts

```diff
@@ -621,6 +674,21 @@ function runFullASTValidation(
               locations: node.location ? [node.location] : [],
             })
           );
+        } else {
+          if (functions.length > 1) {
```

Comment:
I'm not sure whether this is a future-proof way to check - the way our full reference operations are set up, they could theoretically accept multiple parameters. Should we check the `requiredReferences` here?

Reply:
This code is generic enough to check functions in different position in the node, as `moving_average( median( ...), median( ... ) )` as well as `moving_average( median( ...), window = 7, shift="7d", median( ... ) )`.

I could parametrize the threshold to be `requiredReferences.filter( /* check for functions */).length` here, but in general the formula code assumes only one argument to be a reference (to a field or operation) and it may be required to change in the future if we introduce operations with different assumptions.
The parametric change here is pretty easy and I'll push a fix on that.